src/amt/APFProcessor.ts

```diff
@@ -68,92 +69,124 @@ export enum APFChannelOpenFailureReasonCode {
     }
-
+    const cmd = socket.tag.accumulator.charCodeAt(0)
     switch (cmd) {
       case APFProtocol.KEEPALIVE_REQUEST:
-        return APFProcessor.keepAliveRequest(socket, len, data)
+        return APFProcessor.keepAliveRequest(socket, data.length, data)
       case APFProtocol.KEEPALIVE_REPLY:
-        return APFProcessor.keepAliveReply(len)
+        return APFProcessor.keepAliveReply(data.length, data)
       case APFProtocol.KEEPALIVE_OPTIONS_REPLY:
-        return APFProcessor.keepAliveOptionsReply(len, data)
+        return APFProcessor.keepAliveOptionsReply(data.length, data)
       case APFProtocol.PROTOCOLVERSION:
-        return APFProcessor.protocolVersion(socket, len, data)
+        return APFProcessor.protocolVersion(socket, data.length, data)
       case APFProtocol.USERAUTH_REQUEST:
-        return APFProcessor.userAuthRequest(socket, len, data)
+        return APFProcessor.userAuthRequest(socket, data.length, data)
       case APFProtocol.SERVICE_REQUEST:
-        return APFProcessor.serviceRequest(socket, len, data)
+        return APFProcessor.serviceRequest(socket, data.length, data)
       case APFProtocol.GLOBAL_REQUEST:
-        return APFProcessor.globalRequest(socket, len, data)
+        return APFProcessor.globalRequest(socket, data.length, data)
       case APFProtocol.CHANNEL_OPEN:
-        return APFProcessor.channelOpen(socket, len, data)
+        return APFProcessor.channelOpen(socket, data.length, data)
       case APFProtocol.CHANNEL_OPEN_CONFIRMATION:
-        return APFProcessor.channelOpenConfirmation(socket, len, data)
+        return APFProcessor.channelOpenConfirmation(socket, data.length, data)
       case APFProtocol.CHANNEL_OPEN_FAILURE:
-        return APFProcessor.channelOpenFailure(socket, len, data)
+        return APFProcessor.channelOpenFailure(socket, data.length, data)
       case APFProtocol.CHANNEL_CLOSE:
-        return APFProcessor.channelClose(socket, len, data)
+        return APFProcessor.channelClose(socket, data.length, data)
       case APFProtocol.CHANNEL_WINDOW_ADJUST:
-        return APFProcessor.channelWindowAdjust(socket, len, data)
+        return APFProcessor.channelWindowAdjust(socket, data.length, data)
       case APFProtocol.CHANNEL_DATA:
-        return APFProcessor.channelData(socket, len, data)
+        return APFProcessor.channelData(socket, data.length, data)
       case APFProtocol.DISCONNECT:
-        return APFProcessor.disconnect(socket, len, data)
+        return APFProcessor.disconnect(socket, data.length, data)
       default: {
         logger.warn(`${messages.MPS_UNKNOWN_CIRA_COMMAND}: ${cmd}`)
         return -1
       }
     }
   },
 
-  disconnect: (socket: CIRASocket, len: number, data: string): number => {
-    if (len < 7) return 0
-    const ReasonCode: number = Common.ReadInt(data, 1)
-    logger.silly(`${messages.MPS_DISCONNECT}, ${ReasonCode.toString()}`)
+  disconnect: (socket: CIRASocket, dataLen: number, data: string): number => {
```

Comment:
In general, we tend to prefer full non-abbreviated variable name. So instead of `dataLen`, we'd do `dataLength`